processor/metricstransformprocessor/metrics_transform_processor_helper_test.go

```diff
@@ -0,0 +1,633 @@
+					},
+				},
+			},
+			in: []*metricspb.Metric{
+				TestcaseBuilder().SetName("metric1").SetLabels([]string{"label1", "label2"}).SetDataType(metricspb.MetricDescriptor_GAUGE_INT64).
+					AddTimeseries(2, []string{"label1-value1", "label2-value1"}).AddTimeseries(1, []string{"label1-value1", "label2-value2"}).
+					AddInt64Point(0, 3, 2).AddInt64Point(1, 1, 2).
+					Build(),
+			},
+			out: []*metricspb.Metric{
+				TestcaseBuilder().SetName("metric1").SetLabels([]string{"label1", "label2"}).SetDataType(metricspb.MetricDescriptor_GAUGE_INT64).
+					AddTimeseries(2, []string{"label1-value1", "label2-value1"}).AddTimeseries(1, []string{"label1-value1", "label2-value2"}).
+					AddInt64Point(0, 3, 2).AddInt64Point(1, 1, 2).
+					Build(),
+				TestcaseBuilder().SetName("metric1").SetLabels([]string{"label1", "label2"}).SetDataType(metricspb.MetricDescriptor_GAUGE_INT64).
+					AddTimeseries(1, []string{"label1-value1", "new/label2-value"}).
+					AddInt64Point(0, 4, 2).
+					Build(),
+			},
+		},
+		{
+			name: "metric_labels_aggregation_sum_distribution_insert",
+			transforms: []Transform{
+				{
+					MetricName: "metric1",
+					Action:     Insert,
+					Operations: []Operation{
+						{
+							Action:          AggregateLabels,
+							LabelSetMap:     map[string]bool{"label1": true},
+							AggregationType: Sum,
+						},
+					},
+				},
+			},
+			in: []*metricspb.Metric{
+				TestcaseBuilder().SetName("metric1").SetLabels([]string{"label1", "label2"}).SetDataType(metricspb.MetricDescriptor_GAUGE_DISTRIBUTION).
+					AddTimeseries(1, []string{"label1-value1", "label2-value1"}).AddTimeseries(3, []string{"label1-value1", "label2-value2"}).
+					AddDistributionPoints(0, 1, 3, 6, []float64{1, 2}, []int64{0, 1, 2}, 3).
+					AddDistributionPoints(1, 1, 5, 10, []float64{1, 2}, []int64{1, 1, 3}, 4).
+					Build(),
+			},
+			out: []*metricspb.Metric{
+				TestcaseBuilder().SetName("metric1").SetLabels([]string{"label1", "label2"}).SetDataType(metricspb.MetricDescriptor_GAUGE_DISTRIBUTION).
+					AddTimeseries(1, []string{"label1-value1", "label2-value1"}).AddTimeseries(3, []string{"label1-value1", "label2-value2"}).
+					AddDistributionPoints(0, 1, 3, 6, []float64{1, 2}, []int64{0, 1, 2}, 3).
+					AddDistributionPoints(1, 1, 5, 10, []float64{1, 2}, []int64{1, 1, 3}, 4).
+					Build(),
+				TestcaseBuilder().SetName("metric1").SetLabels([]string{"label1"}).SetDataType(metricspb.MetricDescriptor_GAUGE_DISTRIBUTION).
+					AddTimeseries(1, []string{"label1-value1"}).
+					AddDistributionPoints(0, 1, 8, 16, []float64{1, 2}, []int64{1, 2, 5}, 7).
+					Build(),
+			},
+		},
+	}
+)
+
+// constructTestInputMetricsDate builds the actual metrics from the test cases
+func constructTestInputMetricsData(test metricsTransformTest) consumerdata.MetricsData {
+	md := consumerdata.MetricsData{
+		Metrics: make([]*metricspb.Metric, len(test.in)),
+	}
+	for idx, in := range test.in {
+		// compose the metric
+		md.Metrics[idx] = in
+	}
+	return md
```

Comment:
What's wrong with:
```go
return consumerdata.MetricsData{ Metrics: test.in }
```

Reply:
Ah didn't catch this after the refactoring.